src/panorama/scripts/custom_game/game_setup.js

```diff
@@ -2686,52 +2687,52 @@ function OnSkillTabShown(tabName) {
                     if(filterInfo.shouldShow) {
                         if(useSmartGrouping) {
                             var theOwner = abilityHeroOwner[abilityName];
+                            var neutralGroup = flagDataInverse[abilityName].group;
 
-                            if(theOwner != null) {
+                            if(theOwner != null || neutralGroup != null) {
```

Comment:
Lines 2692 and 2694. Since theOwner and neutralGroup is only used in lines 2694, consider moving line 2694 outside and changing the condition to check for groupKey.